File: typescript/src/utils/logger.ts
/**
 * Logger interface for SDK logging
 */
export interface Logger {
  debug(message: string, meta?: any): void;
  info(message: string, meta?: any): void;
  warn(message: string, meta?: any): void;
  error(message: string, meta?: any): void;
}

/**
 * Console-based logger implementation
 * Logs to console with appropriate log levels
 */
export class ConsoleLogger implements Logger {
  private enabled: boolean;

  constructor(enabled: boolean = true) {
    this.enabled = enabled;
  }

  debug(message: string, meta?: any): void {
    if (this.enabled && process.env.NODE_ENV !== 'production') {
      console.debug(`[DCID SDK] ${message}`, meta || '');
    }
  }

  info(message: string, meta?: any): void {
    if (this.enabled) {
      console.info(`[DCID SDK] ${message}`, meta || '');
    }
  }

  warn(message: string, meta?: any): void {
    if (this.enabled) {
      console.warn(`[DCID SDK] ${message}`, meta || '');
    }
  }

  error(message: string, meta?: any): void {
    if (this.enabled) {
      console.error(`[DCID SDK] ${message}`, meta || '');
    }
  }
}

/**
 * No-op logger implementation (disabled logging)
 */
export class NoOpLogger implements Logger {
  debug(): void {}
  info(): void {}
  warn(): void {}
  error(): void {}
}

File: golang/pkg/dcid/logger.go
package dcid

import "fmt"

// Logger interface for SDK logging
type Logger interface {
	// Debug logs a debug message with optional fields
	Debug(msg string, fields map[string]interface{})
	// Info logs an info message with optional fields
	Info(msg string, fields map[string]interface{})
	// Error logs an error message with optional fields
	Error(msg string, fields map[string]interface{})
}

// NoOpLogger is a logger that does nothing (used in production)
type NoOpLogger struct{}

func (l *NoOpLogger) Debug(msg string, fields map[string]interface{}) {}
func (l *NoOpLogger) Info(msg string, fields map[string]interface{})  {}
func (l *NoOpLogger) Error(msg string, fields map[string]interface{}) {}

// ConsoleLogger is a simple console logger (used in development)
type ConsoleLogger struct {
	verbose bool
}

// NewConsoleLogger creates a new console logger
func NewConsoleLogger(verbose bool) *ConsoleLogger {
	return &ConsoleLogger{verbose: verbose}
}

func (l *ConsoleLogger) Debug(msg string, fields map[string]interface{}) {
	if l.verbose {
		l.log("DEBUG", msg, fields)
	}
}

func (l *ConsoleLogger) Info(msg string, fields map[string]interface{}) {
	l.log("INFO", msg, fields)
}

func (l *ConsoleLogger) Error(msg string, fields map[string]interface{}) {
	l.log("ERROR", msg, fields)
}

func (l *ConsoleLogger) log(level, msg string, fields map[string]interface{}) {
	// Simple console logging - can be enhanced with a proper logging library
	fmt.Printf("[%s] %s", level, msg)
	if len(fields) > 0 {
		fmt.Printf(" %+v", fields)
	}
	fmt.Println()
}

